design/protocol-changes/certificate-vote-diffusion.tex

```diff
@@ -78,61 +78,95 @@ \subsection{Vote diffusion mini-protocol}%
 \subsection{Certificate diffusion mini-protocol}%
 \label{sec:certificate-diffusion}
 
-We propose to use the generic object diffusion mini-protocol
-(\cref{sec:object-diffusion-protocol}) instantiated for certificates. In particular, we want:
+\paragraph{Requirements.}
 
-\begin{description}
-\item[\argfont{object}] a certificate
-\end{description}
-
-A few notes:
+Diffusion of certificates is required in the following contexts:
 
-\begin{itemize}
+\begin{enumerate}
 \item
-  We can assume that there is only one block that reaches quorum per round.
-  In particular, if we have a cert for a round, we do not care at all about any other cert in the same round.
+  Primarily, syncing nodes/pools need to be able to obtain historical certificates in order to choose the correct historical chain, even though certificates do not (meaningfully) affect the ledger state.
 
-\item
-  Probably useless when caught-up. We might want to keep it running just for the awkward situation where we really didn’t get all the votes but there is a cert.
-  Most of the time, we should get all the votes, realise there is a quorum, and therefore know to ignore the certs that peer tell us about.
-  %
-  \alex{
-    I wouldn't say "useless" as we fundamentally need it to defend against equivocation attacks.
-
-    Also, there is this other scenario mentioned on Slack that doesn't involve any adversarial activity where we would need to diffuse certs between caught-up nodes:
-    \url{https://moduscreate.slack.com/archives/C06J5CS446A/p1739264734649699?thread_ts=1739260385.010269&cid=C06J5CS446A}
-  }
-
-\item What about reusing the generic object diffusion mini-protocol?
-  (\cref{sec:object-diffusion-protocol})
-  \begin{itemize}
-  \item id = id’ = roundNumber
+  Concretely, nodes bootstrapping via Ouroboros Genesis (to minimize trust assumptions) need to be able to assess the weight of competing historical chains in order to resist adversarial \emph{long-range attacks}.
+  Computing the weight of a chain requires the corresponding certificates indicating which blocks are boosted (potentially multiple times).
```

Comment:
> (potentially multiple times)

We fear this might bring more confusion than clarity at this stage of the document.

Reply:
Good point, this subtlety should not matter in this context, thanks :+1: